docs/extension.rst

```diff
@@ -0,0 +1,223 @@
+  extension manifest.
+
+  If not specified it defaults to application or runtime branch of the
+  extension point.
+
+.. tip::
+  ``versions`` is same as ``version`` but can be used for specifying 
+  multiple versions.
+
+- ``directory`` is the path relative to prefix where everything is 
+  mounted by flatpak. For a runtime this will be relative to ``/usr``
+  and for an application it will be relative to ``/app``.
+
+- ``add-ld-path`` adds a path relative to the extension prefix 
+  ``/app/extension_directory/foo/lib``, so that a library residing there 
+  can be loaded.
+
+- ``merge-dirs: plug-ins;help`` will merge the contents of these 
+  directories from multiple extensions using the same extension point. 
+  For example, if ``/app/extension_directory/foo/plug-ins`` and 
+  ``/app/extension_directory/bar/plug-ins`` exist, they will be merged 
+  and made available at ``/app/extension_directory/plug-ins``. Similarly
+  for ``help``.
+
+- ``subdirectories: true`` allows to have multiple extensions under the
+  same ``directory``. So two extensions like 
+  ``org.flatpak.app.plugin.foo`` and ``org.flatpak.app.plugin.bar`` will 
+  be mounted at ``/app/extension_directory/foo`` and
+  ``/app/extension_directory/bar``.
+
+- ``no-autodownload: true`` will prevent installing all extensions 
+  under this extension point. This is the default.
+
+- ``autodelete: true`` will remove all extensions under this extension 
+  point when the application is removed.
+
+There are other properties like ``download-if``, ``enable-if``, 
+``autoprune-unless`` etc. which are typically not used in applications.
+
+- ``cleanup-commands`` is used to initialise an empty directory for the
+  extension to be loaded. Use this if nothing in your buildsystem
+  creates that directory. If the directory is missing, there will be
+  an error while running the application.
+
+
+This is a typical example of loading an existing extension from the
+Platform or SDK in the application.
+
+.. code-block:: yaml
+
+  id: org.flatpak.cool-app
+  runtime: org.kde.Platform
+  runtime-version: 5.15-23.08
+  sdk: org.kde.Sdk
+  command: foo
+  add-extensions:
+    org.freedesktop.Platform.ffmpeg-full:
+      version: '23.08'
+      directory: lib/ffmpeg
+      add-ld-path: .
+  cleanup-commands:
+    - mkdir -p ${FLATPAK_DEST}/lib/ffmpeg
+
+Extension manifest
+------------------
+
+Once the extension point is defined, an extension like 
+``org.flatpak.app.plugin.foo`` can be created.
+
+This is a typical example of such an extension manifest. The explanation
+is discussed below.
+
+.. code-block:: yaml
+
+  id: org.flatpak.app.plugin.foo
+  branch: '1.0'
+  runtime: org.flatpak.app
+  runtime-version: 'stable'
+  sdk: org.gnome.Sdk//45
+  build-extension: true
+  separate-locales: false
+  appstream-compose: false
+  build-options:
+    prefix: /app/extension_directory/foo
+    env:
+      PATH: /app/extensions/foo/bin:/app/bin:/usr/bin
```

Comment:
```suggestion
    prepend-path: /app/extensions/foo/bin
```

Reply:
it's only necessary if you install some binary part of the build process.

similarly `prepend-pkg-config-path` for .pc files from dependencies.

Reply:
Done, thanks